src/instructions/port.rs

```diff
@@ -80,84 +80,80 @@ impl PortWrite for u32 {
     }
 }
 
-/// A read only I/O port.
-pub struct PortReadOnly<T> {
-    port: u16,
-    phantom: PhantomData<T>,
+mod sealed {
+    pub trait Access {
+        const DEBUG_NAME: &'static str;
+    }
 }
```

Comment:
I wonder if this sealed trait is necessary. Couldn't we just add a normal (zero-sized) `access` field to `PortGeneric` instead of keeping it in `PhantomData`? Then we could just use the normal `Debug` impl of the `ReadOnlyAccess` etc. structs (and gate the Debug impl on `A: Debug`).

Reply:
I tried this, but then you can't have `PortGeneric::new` as there's no way to create the access value.

Reply:
Ah, because applying `Default::default` is not const possible yet?

Reply:
Exactly, it requires the `const_fn_trait_bound` feature.